src/lib.rs

```diff
@@ -4420,6 +4451,99 @@ mod tests {
             Err(Error::CryptoFail)
         );
     }
+
+    #[test]
+    /// Tests that the MAX_STREAMS frame is sent for bidirectional streams.
+    fn stream_limit_update_bidi() {
+        let mut buf = [0; 65535];
+
+        let mut pipe = testing::Pipe::default().unwrap();
+
+        assert_eq!(pipe.handshake(&mut buf), Ok(()));
```

Comment:
This test has an aspect that could be confusing and might break in the future. It relies on some implicit behaviour of `testing::Pipe::default()` to create the client and server with initial max_streams_* with a value of 3.

Reply:
So, I now changed the tests to use a custom config, so at least the case where we change the default values wouldn't affect this. But yeah, there are probably other ways to make these tests better, but for now this should be enough (might do that in a later PR).